### ISSUE TITLE
Using optimizer gives large KL term

### ISSUE THREAD

**Issue by saikrishnarallabandi** [saikrishnarallabandi/compositionality-expts#6]
Aftr epoch  0  Train KL Loss:  4.0685910552845594e+26 Train CE Loss:  7.474030815264613 Val KL Loss:  4.96021764349418e+26  Val CE Loss:  7.47340297397879 Time:  324.20956802368164

vs Naive update:

Aftr epoch  0  Train KL Loss:  71134.24406546808 Train CE Loss:  7.477112519003681 Val KL Loss:  84342.38469062098  Val CE Loss:  7.476956954997839 Time:  289.88534116744995

**Comment by saikrishnarallabandi** [comment]
@berzentine 
so the initial learning rate in naive update is 20. When we use optimizer, its 0.01 if SGD and 0.001 if Adam

**Comment by saikrishnarallabandi** [comment]
Naive:
for p in model.param:
    .add(-lr, p.grad.data)
20

Pytorch:
optimizer.step()
0.001

**Comment by saikrishnarallabandi** [comment]
looks like high LR is the culprit here

@bhavyakarki can you confirm and close this

**Comment by bhavyakarki** [comment]
High LR and optimizer(LR) different to args.LR
